torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,513 @@
+"""MapInWild dataset."""
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import shutil
+from collections.abc import Sequence
+from typing import Any, Callable, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from torch import Tensor
+
+from torchgeo.datasets.geo import NonGeoDataset
+from torchgeo.datasets.utils import (
+    check_integrity,
+    download_url,
+    extract_archive,
+    percentile_normalization,
+)
+
+
+class MapInWild(NonGeoDataset):
+    """MapInWild dataset.
+
+    The `MapInWild <https://arxiv.org/abs/2212.02265>`_ is curated for the task of
+    wilderness mapping on a pixel-level. MapInWild is a multi-modal dataset and
+    comprises various geodata acquired and formed from different RS sensors
+    over 1018 locations: dual-pol Sentinel-1, four-season Sentinel-2 with 10 bands,
+    ESA WorldCover map, and Visible Infrared Imaging Radiometer Suite
+    NightTime Day/Night band. The dataset consists of 8144 images with the shape
+    of 1920 × 1920 pixels. The images are weakly annotated from the
+    World Database of Protected Areas (WDPA).
+
+    Dataset features:
+    * 1018 areas globally sampled from the WDPA
+    * 10-Band Sentinel-2
+    * Dual-pol Sentinel-1
+    * ESA WorldCover Land Cover
+    * Visible Infrared Imaging Radiometer Suite NightTime Day/Night Band
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://ieeexplore.ieee.org/document/10089830
+
+    .. note::
+       This dataset requires the following additional library to be installed:
+
+       * `pandas <https://pypi.org/project/pandas/>`_ to load CSV files
+
+    .. versionadded:: 0.3
```

Comment:
```suggestion
    .. versionadded:: 0.5
```
Your dataset will first be available in TorchGeo 0.5.0 (expected late August/early September)